diff -up bison-3.2.1/data/c.m4.2~ bison-3.2.1/data/c.m4
--- bison-3.2.1/data/c.m4.2~	2018-11-07 08:16:09.000000000 +0100
+++ bison-3.2.1/data/c.m4	2018-11-15 19:43:56.248635315 +0100
@@ -251,6 +251,7 @@ m4_define([b4_attribute_define],
 #endif
 
 #if defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#ifndef __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
@@ -258,6 +259,7 @@ m4_define([b4_attribute_define],
     _Pragma ("GCC diagnostic ignored \"-Wmaybe-uninitialized\"")
 # define YY_IGNORE_MAYBE_UNINITIALIZED_END \
     _Pragma ("GCC diagnostic pop")
+#endif
 #else
 # define YY_INITIAL_VALUE(Value) Value
 #endif
diff -up bison-3.2.1/src/parse-gram.c.2~ bison-3.2.1/src/parse-gram.c
--- bison-3.2.1/src/parse-gram.c.2~	2018-11-09 06:15:03.000000000 +0100
+++ bison-3.2.1/src/parse-gram.c	2018-11-15 19:44:58.383043967 +0100
@@ -423,6 +423,7 @@ typedef short yytype_int16;
 #endif
 
 #if defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#ifndef __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
@@ -430,6 +431,7 @@ typedef short yytype_int16;
     _Pragma ("GCC diagnostic ignored \"-Wmaybe-uninitialized\"")
 # define YY_IGNORE_MAYBE_UNINITIALIZED_END \
     _Pragma ("GCC diagnostic pop")
+#endif
 #else
 # define YY_INITIAL_VALUE(Value) Value
 #endif
