diff -Naur bison-3.5/data/skeletons/c.m4 bison-3.5.tpg/data/skeletons/c.m4
--- bison-3.5/data/skeletons/c.m4	2019-12-06 07:27:55.000000000 +0000
+++ bison-3.5.tpg/data/skeletons/c.m4	2019-12-16 10:57:31.569004668 +0000
@@ -326,6 +326,7 @@
 #endif
 
 #if defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#ifndef __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN                            \
     _Pragma ("GCC diagnostic push")                                     \
@@ -333,6 +334,7 @@
     _Pragma ("GCC diagnostic ignored \"-Wmaybe-uninitialized\"")
 # define YY_IGNORE_MAYBE_UNINITIALIZED_END      \
     _Pragma ("GCC diagnostic pop")
+#endif
 #else
 # define YY_INITIAL_VALUE(Value) Value
 #endif
diff -Naur bison-3.5/src/parse-gram.c bison-3.5.tpg/src/parse-gram.c
--- bison-3.5/src/parse-gram.c	2019-12-11 06:57:34.000000000 +0000
+++ bison-3.5.tpg/src/parse-gram.c	2019-12-16 10:58:27.332004997 +0000
@@ -352,6 +352,7 @@
 #endif
 
 #if defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#ifndef __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN                            \
     _Pragma ("GCC diagnostic push")                                     \
@@ -359,6 +360,7 @@
     _Pragma ("GCC diagnostic ignored \"-Wmaybe-uninitialized\"")
 # define YY_IGNORE_MAYBE_UNINITIALIZED_END      \
     _Pragma ("GCC diagnostic pop")
+#endif
 #else
 # define YY_INITIAL_VALUE(Value) Value
 #endif
