diff -up bison-3.1/data/c.m4.2~ bison-3.1/data/c.m4
--- bison-3.1/data/c.m4.2~	2018-08-25 10:55:03.000000000 +0200
+++ bison-3.1/data/c.m4	2018-08-28 03:05:53.673922125 +0200
@@ -243,6 +243,7 @@ m4_define([b4_attribute_define],
 #endif
 
 #if defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#ifndef __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
@@ -250,6 +251,7 @@ m4_define([b4_attribute_define],
     _Pragma ("GCC diagnostic ignored \"-Wmaybe-uninitialized\"")
 # define YY_IGNORE_MAYBE_UNINITIALIZED_END \
     _Pragma ("GCC diagnostic pop")
+#endif
 #else
 # define YY_INITIAL_VALUE(Value) Value
 #endif
diff -up bison-3.1/src/parse-gram.c.2~ bison-3.1/src/parse-gram.c
--- bison-3.1/src/parse-gram.c.2~	2018-08-27 20:12:09.000000000 +0200
+++ bison-3.1/src/parse-gram.c	2018-08-28 03:05:02.209740414 +0200
@@ -429,6 +429,7 @@ typedef short yytype_int16;
 #endif
 
 #if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#ifndef __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
@@ -436,6 +437,7 @@ typedef short yytype_int16;
     _Pragma ("GCC diagnostic ignored \"-Wmaybe-uninitialized\"")
 # define YY_IGNORE_MAYBE_UNINITIALIZED_END \
     _Pragma ("GCC diagnostic pop")
+#endif
 #else
 # define YY_INITIAL_VALUE(Value) Value
 #endif
